contractcourt/htlc_success_resolver.go

```diff
@@ -39,19 +37,15 @@ type htlcSuccessResolver struct {
 	// historical queries to the chain for spends/confirmations.
 	broadcastHeight uint32
 
-	// payHash is the payment hash of the original HTLC extended to us.
-	payHash lntypes.Hash
-
 	// sweepTx will be non-nil if we've already crafted a transaction to
 	// sweep a direct HTLC output. This is only a concern if we're sweeping
 	// from the commitment transaction of the remote party.
 	//
 	// TODO(roasbeef): send off to utxobundler
 	sweepTx *wire.MsgTx
 
-	// htlcAmt is the original amount of the htlc, not taking into
-	// account any fees that may have to be paid if it goes on chain.
-	htlcAmt lnwire.MilliSatoshi
+	// htlc contains information on the htlc that we are resolving on-chain.
+	htlc *channeldb.HTLC
```

Comment:
should we copy the htlc instead, i.e. no pointer, so that it can't be mutated externally?

Reply:
I don't know. It is a safety precaution at the cost of a very minimal performance degradation, if any. But just removing the pointer still doesn't make it a deep copy. Also, `channeldb.HTLC` isn't a struct that is typically modified afaik. The most likely bug here would be binding in a range. That would be disastrous, but it is also caught by the linter. I tend towards not defensively dereferencing, but interested you hear ideas

Reply:
It would at least let us avoid the awkward loop binding hack I think

Reply:
> But just removing the pointer still doesn't make it a deep copy

Everything on `channeldb.HTLC` is shallow besides the `Sig` field, which idt we use in the cnct atm. Only brought it up bc as you say, a range binding bug could be pretty disastrous in this area, and seemed like an easy way to avoid the risk altogether.

Reply:
Ok, will remove the pointer then. Cheap insurance

Reply:
Done